mobify/platform-scaffold
# [Checkout Schema] fix saved addresses fetch & update saved address in schema

This PR moves the merlins saved address fetch into the merlins connector, fixes some bugs related to submitting shipping with a saved address and moves saved addresses into the "storedAddresses" key in the store. 

 **JIRA**: https://mobify.atlassian.net/browse/WEBDATA-69
 **Linked PRs**: n/a

## Changes
- Moved saved address fetch into merlins connector
- Called saved address fetch command in initCheckoutShipping command
- Moved logic to get the users saved address details to the submit shipping action, not merlins command
- Renamed the saved address key in the checkout store to "storedAddresses"

## How to test-drive this PR
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- login as a user with saved addresses (eg. mobifyqa@gmail.com)
- add an item to your cart
- proceed through the checkout (using a saved address on the shipping page)
- log out and try placing an order as an unknown users (it should still work)
- your order should be placed correctly
- switch to the SFCC connector
- preview the [SFCC Site](https://preview.mobify.com/?url=https%3A%2F%2Fmobify-tech-prtnr-na03-dw.demandware.net%2Fon%2Fdemandware.store%2FSites-2017refresh-Site%2Fdefault%2FHome-Show&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- login to the site
- add an item to your cart and proceed through the checkout
*saved addresses have not been implemented in the SFCC connector yet, but you should still be able to checkout after filling in correct details*
